### ISSUE TITLE
[Warrior] Shield Wall Removed When Switching Weapons

### ISSUE THREAD

**Issue by JJGodX** [gamefreedomgit/Frostmourne#1901]
[//]: # (REMBEMBER! Add links to things related to the bug using for example:)
[//]: # (https://www.wowhead.com/wotlk)

**Description:**
When switching from a one-handed weapon and shield to a two-handed weapon, Shield Wall is removed.

**How to reproduce:**
Equip a one-handed weapon and shield, cast Shield Wall, and then switch to a two-handed weapon.

**How it should work:**
Shield Wall should remain active and not be removed when switching weapons.
https://www.mmo-champion.com/threads/951897-Shield-Wall-Buff-Stays-On-even-after-requipping-DW-weapons

**Database links:**
https://www.wowhead.com/wotlk/spell=871/shield-wall

**Comment by Piootrek** [comment]
Fixed, I will close this issue once the fix comes live.

**Comment by Manpower99** [comment]
Not sure how this will be fixed, but it is part of a blizzlike bug that got replicated into classic/tbc/wotlk/cata and other private servers too.
Essentially if you unequip a shield the buff gets removed, however if you swap the shield with something else into its offhand, blizzlike scripting never had a check to keep the buff if you still have a shield equipped or not, but rather if you have ANYTHING in offhand equipped.
So essentially the way the bug works is this
1H+Shield > Use Shield wall > Replace shield with a weapon or offhand > Buff remains.
But if you swap into a Two Hander, since there is nothing in offhand the Buff should be removed.
I will be happy to provide proof if required please dont introduce bogus fixes.

**Comment by Piootrek** [comment]
> Not sure how this will be fixed, but it is part of a blizzlike bug that got replicated into classic/tbc/wotlk/cata and other private servers too. Essentially if you unequip a shield the buff gets removed, however if you swap the shield with something else into its offhand, blizzlike scripting never had a check to keep the buff if you still have a shield equipped or not, but rather if you have ANYTHING in offhand equipped. So essentially the way the bug works is this 1H+Shield > Use Shield wall > Replace shield with a weapon or offhand > Buff remains. But if you swap into a Two Hander, since there is nothing in offhand the Buff should be removed. I will be happy to provide proof if required please dont introduce bogus fixes.

Thank you for clarifying, please provide proof anyway.

**Comment by Manpower99** [comment]
https://youtu.be/EwLOcI4h_kg 
 27:05 he uses shield wall and spell reflect and then he has to swap to twohander and when he does the buffs who have this check are removed.
referring to what i said previously, the blizzlike dual wield related bug to shield wall does not apply if you have nothing in offhand.
In this case it was a warrior without titansgrip so a twohander can only be present in mainhand and thus shield wall buff is removed due to nothing being in offhand.
https://www.youtube.com/watch?v=EwLOcI4h_kg&t=1064s another one, he swaps to twohander and thus nothing in offhand to check if theres anything in Offhand and buff is removed.

This is an interaction that existed since vanilla and classic has replicated it in Vanilla Classic/Wotlk Classic/TBC Classic and even cata right now.

Classic dev comment on this :
![image](https://github.com/user-attachments/assets/4ae9b472-4f15-45a2-a019-574e17a1e522)

**Comment by Manpower99** [comment]
Tldr: shield abilities require shield equipped to be usable yes, but if you swap anything to your offhand the buff remains, if you have nothing equipped in offhand the buff is removed.

**Comment by Manpower99** [comment]
https://www.wowhead.com/spell=871/shield-wall#comments:id=1644461 proof the interaction remained all the way till cata

**Comment by Piootrek** [comment]
So, it works correctly without my fix. Reverting.
